crates/oxc_parser/src/lexer/trivia_builder.rs

```diff
@@ -29,3 +29,8 @@ impl TriviaBuilder {
         self.trivias.add_comment(Span::new(start + 2, end - 2), CommentKind::MultiLine);
     }
 }
+
+pub fn is_eslint_configuration_comment(comment: &str) -> bool {
```

Comment:
I think we can create a new struct inside the linter to hold all eslint related info, the lexer / parser doesn't need to know about eslint.